test/integration/01-setup/01-funds-bitcoind-lnds.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { baseLogger } from "src/logger"
+import { btc2sat } from "src/utils"
+import { MainBook } from "src/mongodb"
+import { liabilitiesReserve, lndAccountingPath } from "src/ledger/ledger"
+import {
+  checkIsBalanced,
+  lnd1,
+  lndOutside1,
+  bitcoindClient,
+  getChainBalance,
+  fundLnd,
+  waitUntilSyncAll,
+} from "test/helpers"
+
+jest.mock("src/realtimePrice", () => require("test/mocks/realtimePrice"))
+
+const defaultWallet = ""
+
+beforeEach(async () => {
+  await waitUntilSyncAll()
+})
+
+describe("Bitcoind", () => {
+  it("should be funded mining 10 blocks", async () => {
+    const numOfBlock = 10
+    try {
+      const { name } = await bitcoindClient.createWallet(defaultWallet)
+      // depends of bitcoind version. needed in < 0.20 but failed in 0.21?
+      expect(name).toBe(defaultWallet)
+    } catch (error) {
+      baseLogger.warn({ error }, "bitcoind wallet already exists")
+    }
+
+    const bitcoindAddress = await bitcoindClient.getNewAddress()
+    await bitcoindClient.mineAndConfirm(numOfBlock, bitcoindAddress)
+    const balance = await bitcoindClient.getBalance()
+
+    // invalid only when there are problems deleting channels and
+    // they get confirmed while mineAndConfirm (101 blocks)
+    expect(balance).toBeGreaterThanOrEqual(50 * numOfBlock)
+  })
+
+  it("funds outside lnd node", async () => {
+    const amount = 1
+    const { chain_balance: initialBalance } = await getChainBalance({ lnd: lndOutside1 })
+    const sats = initialBalance + btc2sat(amount)
+    await fundLnd(lndOutside1, amount)
+    const { chain_balance: balance } = await getChainBalance({ lnd: lndOutside1 })
+    expect(balance).toBe(sats)
+  })
+
+  it("funds lnd1 node", async () => {
+    const amount = 1
+    const { chain_balance: initialBalance } = await getChainBalance({ lnd: lnd1 })
+    const sats = initialBalance + btc2sat(amount)
+    await fundLnd(lnd1, amount)
+    const { chain_balance: balance } = await getChainBalance({ lnd: lnd1 })
+
+    expect(balance).toBe(sats)
+
+    const metadata = { type: "onchain_receipt", currency: "BTC", pending: "false" }
+
+    await MainBook.entry("funding tx")
```

Comment:
this tweak is no longer necessary. we can just used getOnchainAddress from lnd and pay directly to that address

the reason I used this tweak before is because I experience where there could be 3 internal lnd, and the first 2 would only have lightning capability, not onchain capability. but this is no longer the case. not lnd1 has onchain capability so we don't need to explicitly add some funds in the books.

Reply:
uhmm if I remove the code checkIsBalanced fail... also, onchainTransactionEventHandler cant be used to balance the book because it needs an associated user

Reply:
yes because here you're creating a new address directly by calling `getChainBalance` from lnd. if you were to create a new address using the UserWallet class, then the received funds would be associated with the User and there would be no more accounting gap

it's basically what we're doing on the receive onchain test:

```
describe("FunderWallet - On chain", () => {
  it("receives on-chain transaction", async () => {
    const funderWallet = await getFunderWallet({ logger: baseLogger })
    await sendToWallet({ walletDestination: funderWallet })
  })
})
```

Reply:
No, after lnd funding, the test fail in 01-funds-bitcoind-lnds.spec.ts

Reply:
this was the previous test https://github.com/GaloyMoney/galoy/blob/1bcf8b11ff30030c786aa4a671bea69443de06f9/test/integration/02b-onchain-receive.spec.ts#L154 really I cant find the trigger that handle the payment for the first fund transaction to lnd (has not associated user)

Reply:
oh ok the user are creating later now?

Reply:
yup this is true that this remove the user dependancy. I have not thought about it. ok let's close this.

Reply:
is the first btc transaction to be able to create channels, really not sure how is balanced in production because I cant find where it handles the transactions not associated with users.

Please check https://github.com/GaloyMoney/galoy/blob/main/src/entrypoint/trigger.ts#L102 it just return when no user is associated

Reply:
if you like I can just receive onchain from funder user but... when you move onchain txs to bitcoind how will you fund/initialize the lnds?

Reply:
> is the first btc transaction to be able to create channels, really not sure how is balanced in production because I cant find where it handles the transactions not associated with users.

we (typically) don't have transaction not associated with users. nor we should I believe. so this is more a test related stuff.

Reply:
> if you like I can just receive onchain from funder user but... when you move onchain txs to bitcoind how will you fund/initialize the lnds?

not sure to get your question

Reply:
Fixed https://github.com/GaloyMoney/galoy/pull/340/commits/86e070819b14741b3c5033c5156b2259b7358f23 but @jotapea please keep in mind

if on chain transactions are moved from lnds to bitcoind then there is no way to fund (and register in balance sheet) the funds sent to lnds (to open channels).

Reply:
this is a good point... but we can funds directly a channel from a cold storage/bitcoind wallet. 

using the open function from [bos](https://github.com/alexbosworth/balanceofsatoshis ) as a command line tool now to do it but we could integrate this into our codebase. (bos#open is a wrapper above lnd api to make this possible)